CommonMark/Parser/InlineMethods.cs

```diff
@@ -27,7 +28,7 @@ internal static class InlineMethods
             p['_'] = HandleEmphasis;
             p['*'] = HandleEmphasis;
             p['['] = HandleLeftSquareBracket;
-            p[']'] = HandleRightSquareBracket;
+            p[']'] = subj => HandleRightSquareBracket(subj, placeholderBracket);
```

Comment:
It would have to be tested, however it feels like 

```
if (placeholderBracket)
    p[']'] = subj => HandleRightSquareBracket(subj, true);
else
    p[']'] = subj => HandleRightSquareBracket(subj, false);
```

could be optimized better by the compiler since in this case the anonymous function is basically static - it does not need to capture the scope context.